src/lib/array.jl

```diff
@@ -124,6 +124,13 @@ end
   end
 end
 
+@adjoint function repeat(x::AbstractVecOrMat, m::Integer, n::Integer=1)
+   size₁, size₂ = size(x, 1), size(x, 2)
+   begin₁, begin₂ = firstindex(x, 1), firstindex(x, 2)
+   end₁,   end₂   =  lastindex(x, 1),  lastindex(x, 2)
+   return repeat(x, m, n), ȳ -> (sum(@view ȳ[(begin₁ + i*size₁):(end₁ + i*size₁), (begin₂ + j*size₂):(end₂ + j*size₂)] for i ∈ 0:(m-1), j ∈ 0:(n-1)), nothing, nothing)
```

Comment:
```suggestion
   return repeat(x, m, n), ȳ -> begin
      ȳ = reshape(ȳ, size(x,1), m, size(x,2), n)
      dropdims(sum(ȳ, dims=(2,4)), dims=(2,4))
   end
```

Reply:
This (like the original version) will return matrix for vector input, so `AbstractVector` needs a separate method

Reply:
`reshape(sum(ȳ, dims=(2,4)), size(x))` will produce vector output for vector input, so this version should be fine for VecOrMat

Reply:
Indeed. Thanks, done

Reply:
This will still produce a matrix for a vector input in the forward pass. So looks like we do need a separate method